frontend/src/metabase/dashboard/components/DashCardCardParameterMapper.jsx

```diff
@@ -39,6 +39,36 @@ function formatSelected({ name, sectionName }) {
   return `${sectionName}.${name}`;
 }
 
+function getSelectedMappingOption(target, mappingOptions) {
```

Comment:
Fix for retrieving selected mapping option as sometimes the `target` will be of the form:
```
["dimension", ["field", "Custom column"]
```
but the `mapping target` will be:
```
["dimension", ["field", "Custom column", null]
```

So we are reformatting the targets with only the relevant data if the field parts do not match.

__UI__

When a filter has been associated with a custom column, when editing the filter, the custom column in the card is not selected for the filter.
Fixing: the associated custom column will be selected when editing the filter.

Reply:
Looks like the root of this is https://github.com/metabase/metabase/blob/41c7897dfef1b8db37c71218a73ca7e020954393/frontend/src/metabase-lib/lib/Dimension.ts#L1201 when it is called when generating the mapping options here https://github.com/metabase/metabase/blob/41c7897dfef1b8db37c71218a73ca7e020954393/frontend/src/metabase/parameters/utils/mapping-options.js#L16 which differs from what the endpoint returns -- eq `["expression", "foo"]` and not `["expression", "foo", null]`

This works fine, I think, but let's move it to a utils file because it is introspective of mapping and target data structures, and I'd prefer if we avoided that sort of logic in UI files -- I think `frontend/src/metabase/parameters/utils/targets.js` or `frontend/src/metabase/parameters/utils/mapping-options.js` both are a good place for this function.